Framework/Algorithms/src/CalcCountRate.cpp

```diff
@@ -0,0 +1,653 @@
+#include "MantidAlgorithms/CalcCountRate.h"
+
+#include "MantidKernel/PropertyWithValue.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/TimeSeriesProperty.h"
+#include "MantidKernel/ListValidator.h"
+#include "MantidKernel/UnitFactory.h"
+#include "MantidKernel/MandatoryValidator.h"
+#include "MantidKernel/make_unique.h"
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/NumericAxis.h"
+#include "MantidAPI/WorkspaceFactory.h"
+
+#include "MantidDataObjects/Workspace2D.h"
+#include <numeric>
+
+namespace Mantid {
+namespace Algorithms {
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(CalcCountRate)
+
+//----------------------------------------------------------------------------------------------
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string CalcCountRate::name() const { return "CalcCountRate"; }
+
+/// Algorithm's version for identification. @see Algorithm::version
+int CalcCountRate::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string CalcCountRate::category() const {
+  return "Inelastic\\Utility";
```

Comment:
Diagnostics good, events..., may be filtering, though not exactly. Ok, add it to filtering -- there somebody looking for spurion filtering may want to look and something like RebinBySampleTime is already sitting.